layer/nrlf/core/validators.py

```diff
@@ -316,3 +317,64 @@ def _validate_ssp_asid(self, model: DocumentReference):
                 field=f"context.related[{idx}].identifier.value",
             )
             return
+
+    def _validate_category(self, model: DocumentReference):
+        """
+        Validate the category field contains an appropriate coding system, code and display.
+        """
+
+        if len(model.category) > 1:
+            logger.log(
+                LogReference.VALIDATOR001, step="category", reason="category_too_long"
+            )
+            self.result.add_error(
+                issue_code="invalid",
+                error_code="INVALID_RESOURCE",
+                diagnostics=f"Invalid category length: {len(model.category)}",
+                field=f"category",
+            )
+            return
+
+        logger.log(LogReference.VALIDATOR001, step="category")
+
+        logger.debug("Validating category")
+
+        for index, coding in enumerate(model.category[0].coding):
+            if coding.system != "http://snomed.info/sct":
+                self.result.add_error(
+                    issue_code="value",
+                    error_code="INVALID_RESOURCE",
+                    diagnostics=f"Invalid category system: {coding.system}",
+                    field=f"category[0].coding[{index}].system",
+                )
+                continue
+
+            if coding.code not in [
+                "734163000",
+                "1102421000000108",
+            ]:
+                self.result.add_error(
+                    issue_code="value",
+                    error_code="INVALID_RESOURCE",
+                    diagnostics=f"Invalid category code: {coding.code}",
```

Comment:
Could add info about what's expected too, maybe something like `Category code must be '734163000' or '1102421000000108'.`